File: blipapi_avatar.php
<?php

/**
 * Blip! (http://blip.pl) communication library.
 *
 * @author Marcin Sztolcman <marcin /at/ urzenia /dot/ net>
 * @version 0.02.13
 * @version $Id$
 * @copyright Copyright (c) 2007, Marcin Sztolcman
 * @license http://opensource.org/licenses/gpl-license.php GNU Public License v.2
 * @package blipapi
 */

/**
 * Blip! (http://blip.pl) communication library.
 *
 * @author Marcin Sztolcman <marcin /at/ urzenia /dot/ net>
 * @version 0.02.13
 * @version $Id$
 * @copyright Copyright (c) 2007, Marcin Sztolcman
 * @license http://opensource.org/licenses/gpl-license.php GNU Public License v.2
 * @package blipapi
 */

if (!class_exists ('BlipApi_Avatar')) {
    class BlipApi_Avatar implements IBlipApi_Command {
        /**
        * Get info about users avatar
        *
        * @param string $user
        * @access public
        * @return array parameters for BlipApi::__query
        */
        public static function read ($user) {
            if (!$user) {
                throw new UnexpectedValueException ('User name is missing.', -1);
            }
            return array ("/users/$user/avatar", 'get');
        }

        /**
        * Upload new avatar
        *
        * Throws UnexpectedValueException if avatar path is missing or file not found
        *
        * @param string $avatar new avatars path
        * @access public
        * @return array parameters for BlipApi::__query
        */
        public static function update ($avatar) {
            if (!$avatar || !file_exists ($avatar)) {
                throw new UnexpectedValueException ('Avatar path missing or file not found.', -1);
            }
            if ($avatar[0] != '@') {
                $avatar = '@'.$avatar;
            }
            return array ('/avatar', 'post', array ( 'avatar[file]' => $avatar ), array ('multipart' => 1));
        }

        /**
        * Delete avatar
        *
        * @access public
        * @return array parameters for BlipApi::__query
        */
        public static function delete () {
            return array ('/avatar', 'delete');
        }
    }
}

File: blipapi_avatar.py
# -*- coding: utf-8 -*-
# $Id$
#
# Blip! (http://blip.pl) communication library.
# Author: Marcin Sztolcman <marcin /at/ urzenia /dot/ net>
# Version: 0.02.04
# Copyright: (r) 2009 Marcin Sztolcman
# License: http://opensource.org/licenses/gpl-license.php GNU Public License v.2

import os.path

from blipapi__utils import prepare_post_field

def read (user):
    """ Get info about specified user's avatar """
    if not user:
        raise ValueError ('User name is missing.')
    return ['/users/' + str (user) + '/avatar', 'get', None, None]

def update (avatar):
    """ Update current user avatar """
    if not os.path.isfile (avatar):
        raise ValueError ('Avatar path missing or file not found.')

    opts = dict ()

    multipart, boundary = prepare_post_field ('avatar[file]', avatar, is_file=True)

    opts['multipart']   = multipart
    opts['boundary']    = boundary

    return ('/avatar', 'post', None, opts)

def delete ():
    """ Delete current user avatar """
    return ('/avatar', 'delete', None, None)

